File: AndroidJava/Java/app/src/main/java/com/paytapsintegrationjava/PaymentDetailActivity.java
package com.paytapsintegrationjava;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;
import androidx.recyclerview.widget.LinearLayoutManager;

import com.paytapsintegrationjava.adpter.PaymentDetailAdapter;
import com.paytapsintegrationjava.databinding.ActivityPaymentDetailBinding;
import com.paytapsintegrationjava.model.CommonModel;
import com.paytapsintegrationjava.model.ResponseModel;

import java.util.ArrayList;

public class PaymentDetailActivity extends AppCompatActivity {

    ActivityPaymentDetailBinding view;
    Context context = this;
    ArrayList<CommonModel> arrayList = new ArrayList<>();
    ResponseModel model;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        view = DataBindingUtil.setContentView(this, R.layout.activity_payment_detail);

        init();
    }

    private void init() {
        getIntentData();
        setOnClickListeners();
    }

    /**
     * This method is used to set click listeners on views
     */
    private void setOnClickListeners() {
        view.ivBack.setOnClickListener(v -> {
            onBackPressed();
        });
    }

    /***
     * This method is used to get data from the previous activity
     */
    private void getIntentData() {
        Intent in = getIntent();
        model = (ResponseModel) in.getSerializableExtra("model");
        if (model != null) {
            arrayList.add(new CommonModel("Order ID:-", model.getOrderID()));
            arrayList.add(new CommonModel("Transaction Id:-", model.getTransactionId()));
            arrayList.add(new CommonModel("Token:-", model.getToken()));
            arrayList.add(new CommonModel("Amount:-", model.getAmount()));
            arrayList.add(new CommonModel("Transaction by:-", model.getTransactionTitle()));
            arrayList.add(new CommonModel("Email:-", model.getEmail()));
            arrayList.add(new CommonModel("Number:-", model.getNumber()));

            view.recyclerView.setLayoutManager(new LinearLayoutManager(context, LinearLayoutManager.VERTICAL, false));
            // Set the data in adapter
            view.recyclerView.setAdapter(new PaymentDetailAdapter(context, arrayList));

        }
    }
}

File: AndroidKotlin/Kotlin/app/src/main/java/com/paytapsintegrationkotlin/PaymentDetailActivity.kt
package com.paytapsintegrationkotlin

import android.content.Context
import android.content.Intent
import android.os.Bundle

import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import com.paytapsintegrationkotlin.adpter.PaymentDetailAdapter


import com.paytapsintegrationkotlin.model.CommonModel
import com.paytapsintegrationkotlin.model.ResponseModel
import kotlinx.android.synthetic.main.activity_payment_detail.*

import java.util.ArrayList

class PaymentDetailActivity : AppCompatActivity() {

    internal var context: Context = this
    internal var arrayList = ArrayList<CommonModel>()
    internal var model: ResponseModel? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_payment_detail)

        init()
    }

    private fun init() {
        getIntentData()
        setOnClickListeners()
    }

    /**
     * This method is used to set click listeners on views
     */
    private fun setOnClickListeners() {
        ivBack.setOnClickListener { v -> onBackPressed() }
    }


    /***
     * This method is used to get data from the previous activity
     */
    private fun getIntentData() {
        val `in` = intent
        model = `in`.getSerializableExtra("model") as ResponseModel
        if (model != null) {
            arrayList.add(CommonModel("Order ID:-", model!!.orderID))
            arrayList.add(CommonModel("Transaction Id:-", model!!.transactionId))
            arrayList.add(CommonModel("Token:-", model!!.token))
            arrayList.add(CommonModel("Amount:-", model!!.amount))
            arrayList.add(CommonModel("Transaction by:-", model!!.transactionTitle))
            arrayList.add(CommonModel("Email:-", model!!.email))
            arrayList.add(CommonModel("Number:-", model!!.number))

            recyclerView.setLayoutManager(
                LinearLayoutManager(
                    context,
                    LinearLayoutManager.VERTICAL,
                    false
                )
            )
            // Set the data in adapter
            recyclerView.setAdapter(PaymentDetailAdapter(context, arrayList))
        }
    }
}
